### PR TITLE
Api dev

### PR DESCRIPTION
**Pull request by sarahswinton**
- Major refactoring of API. 
- Addition of Postman testing. 
- Integration with GitHub actions.

### CODE REVIEW

File: api/routes/apiRoutes.js [hunk 1]
```diff
@@ -4,121 +4,137 @@ module.exports = function(app) {
...
     * @returns {Error}  default - Unexpected error
     */
-
-    app.route('/state/user-home')
-        .put(state.update_user_home);
     
+    /**
+    * This function comment is parsed by doctrine
+    * @route DELETE /userHome/user-home
+    * @group state - Operations about system states
```
Comment (iwishiwasaneagle) [inline api/routes/apiRoutes.js:109]: No mention of `:userHomeId` in docs

File: api/routes/apiRoutes.js [hunk 2]
```diff
@@ -4,121 +4,137 @@ module.exports = function(app) {
...
 
-    app.route('/state/present-mask')
-        .get(state.read_mask_present)
-        .put(state.update_mask_present);
+    app.route('/state/mask/latest')
+        .get((res,req)=>state.get_latest_status(res,req,'mask'));
 
     /**
     * This function comment is parsed by doctrine
-    * @route GET /state/door-switch
```
Comment (iwishiwasaneagle) [inline api/routes/apiRoutes.js:73]: Explain what `:statusId` is in doc

**Review by iwishiwasaneagle — CHANGES_REQUESTED** [review]

File: api/routes/apiRoutes.js (hunk 1 above)
Comment (sarahswinton) [reply api/routes/apiRoutes.js:114]: Good catch. This has been resolved in the most recent push.

File: api/routes/apiRoutes.js (hunk 2 above)
Comment (sarahswinton) [reply api/routes/apiRoutes.js:73]: This has been resolved in the most recent push.

**Pull request merged**